ludwig/datasets/base_dataset.py

```diff
@@ -0,0 +1,86 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import abc
+import os
+from pathlib import Path
+import yaml
+import pandas as pd
+
+
+"""An abstract base class that defines a set of methods for download,
+preprocess and plug data into the ludwig training API"""
+
+
+class BaseDataset(metaclass=abc.ABCMeta):
+    def __init__(self, dataset_name):
```

Comment:
Cache location should be configured by the user as opposed to the dataset.  Let's make it a keyword arg:

```
def __init__(self, dataset_name, cache=DEFAULT_CACHE_DIR)
```

Then we can define the default cache constant above the class definition in this file:

```
DEFAULT_CACHE_DIR = str(Path.home().joinpath('.ludwig_cache'))
```

Reply:
Added